.gitignore

```diff
@@ -3,6 +3,7 @@ target/
 !.mvn/wrapper/maven-wrapper.jar
 !**/src/main/**
 !**/src/test/**
+nerfums-app/node_modules
```

Comment:
one thing to note, is that there are usually some default .gitignore files for various project types. For example, angular has one here: https://github.com/angular/angular/blob/master/.gitignore

Java, .NET, etc will all have their own defaults that automatically ignore build files, external libraries, etc